### PR TITLE
egeria#1778 - Expose unique message id to message consumers

### PR DESCRIPTION
**Pull request by jnhagelberg**
See https://github.com/odpi/egeria/issues/1778

### CODE REVIEW

**Comment by jnhagelberg** [comment]
@mandy-chessell or @planetf1 can you please take a look at this?

File: open-metadata-implementation/repository-services/repository-services-apis/src/main/java/org/odpi/openmetadata/repositoryservices/connectors/omrstopic/InternalOMRSEventProcessingContext.java [hunk 1]
```diff
@@ -77,4 +80,23 @@ public static void clear() {
         INSTANCE.set(null);
         
     }
+    /**
+     * Gets the unique identifier for the current message that
+     * is being processed.
+     * 
+     * @return
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-apis/src/main/java/org/odpi/openmetadata/repositoryservices/connectors/omrstopic/InternalOMRSEventProcessingContext.java]: add description of return value

File: open-metadata-implementation/repository-services/repository-services-apis/src/main/java/org/odpi/openmetadata/repositoryservices/connectors/omrstopic/InternalOMRSEventProcessingContext.java [hunk 2]
```diff
@@ -77,4 +80,23 @@ public static void clear() {
...
+     */
+    public String getCurrentMessageId() {
+        return currentMessageId;
+    }
+    
+    /**
+     * Sets the unique identifier for the current message that
+     * is being processed.
+     * 
+     * @param messageId
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-apis/src/main/java/org/odpi/openmetadata/repositoryservices/connectors/omrstopic/InternalOMRSEventProcessingContext.java]: Add parameter description

File: open-metadata-implementation/repository-services/repository-services-apis/src/main/java/org/odpi/openmetadata/repositoryservices/connectors/omrstopic/OMRSEventProcessingContext.java [hunk 3]
```diff
@@ -40,4 +40,18 @@ public static void addAsyncProcessingResult(Future<?> future) {
...
+    
+    /**
+     * Gets unique identifier for the current message.  The identifier is specific to
+     * the underlying messaging system.  If a message is being reprocessed (eg due
+     * to a server restart before the message was fully processed)(, the identifier
+     * will be the same.  The identifier should be different if content happens
+     * to be the same as the content of another message but they are actually different
+     * messages.
+     * 
+     * @return
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-apis/src/main/java/org/odpi/openmetadata/repositoryservices/connectors/omrstopic/OMRSEventProcessingContext.java]: Add return value description

**Comment by jnhagelberg** [comment]
@mandy-chessell Thanks for looking at this.  I have pushed changes to address your review comments.

**Comment by jnhagelberg** [comment]
@mandy-chessell FYI, I just rebased the changes to resolve the merge confilcts.

**Comment by jnhagelberg** [comment]
rebased again

**Comment by planetf1** [comment]
Unfortunate for all the rebasing -- plenty going in esp with dependencies (by fault!) so merged you quickly this morning :-) !

**Pull request merged**